test/e2e/activator_test.go

```diff
@@ -0,0 +1,176 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"testing"
+	"time"
+	"sync/atomic"
+	"net/http"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/knative/pkg/test/logging"
+	"github.com/knative/pkg/test/spoof"
+	"github.com/knative/serving/test"
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	pkgTest "github.com/knative/pkg/test"
+)
+
+// TestActivatorOverload makes sure that activator can handle the load when scaling from 0.
+// We need to add a similar test for the User pod overload once the second part of overload handling is done.
+func TestActivatorOverload(t *testing.T) {
+	var (
+		logger  = logging.GetContextLogger(t.Name())
```

Comment:
This may not cause problems with `-race` due to the fact it's the only test using this, but you might consider `t.Logf` to avoid the races involved with `logging.GetContextLogger`

Reply:
Yes, it seems this is the only e2e test using the logger (there are some performance tests using it though). I'd vote for keeping this logger for now to let the test bake, since it provides us streaming and the timestamps. We could replace it later with `t.Logf` to make it on-par with other tests.

Reply:
Does that mean that this is the only test that is able to "use this while baking"? If we add more we'll get random races?

Reply:
Ok, losing with the score is 1:2 :)
I can replace the logger, it's the trade off between readability and the potential racing if someone adds a new test with the logger in the suite.